FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.DataClient.UnitTests/Api/AzureAccessTokenProviderTests.cs

```diff
@@ -28,9 +30,24 @@ public async Task GivenAnInvalidResourceUrl_WhenGetAccessToken_ArgumentException
         [InlineData(null)]
         public async Task GivenANullResourceUrl_WhenGetAccessToken_ArgumentNullExceptionShouldBeThrown(string resourceUrl)
         {
-            var accessTokenProvider = new AzureAccessTokenProvider(new NullLogger<AzureAccessTokenProvider>());
+            var accessTokenProvider = new AzureAccessTokenProvider(new ExternalTokenCredentialProvider(new NullLogger<ExternalTokenCredentialProvider>()), new NullLogger<AzureAccessTokenProvider>());
```

Comment:
What's the consideration to use `ExternalTokenCredentialProvider` instead of `MockTokenCredentialProvider` here?

Reply:
Updated these unit tests.